internal/service/client.go

```diff
@@ -15,8 +15,14 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
-func grpcClient(ctx *context.Context) (*grpc.ClientConn, *context.Context, error) {
+func grpcClient(ctx *context.Context, traceIDOptional ...string) (*grpc.ClientConn, *context.Context, error) {
 	appConfig := config.GetConfig()
+
```

Comment:
Can we generate `traceId` here itself? That way we can avoid writing it in all the commands

Reply:
Every command creates its own copy of the client, and the client is not responsible for sending errors. Basically, if we generate a trace ID in the client, we will have to return it with the client context for the connection-creating method to log it with an error. Rather, this should be the responsibility of the method using the client to pass and log the trace ID (if required) and ignore the field if this is of no use.

Reply:
See if we can use client grpc interceptors to add trace id in the outgoing request so that we can avoid trace id login in each command (may be missed in some commands)

Reply:
Done